src/Processors/Formats/Impl/Parquet/RowGroupChunkReader.h

```diff
@@ -0,0 +1,140 @@
+#pragma once
+#include <Processors/Formats/Impl/Parquet/SelectiveColumnReader.h>
+#include <Processors/Formats/Impl/Parquet/ParquetColumnReaderFactory.h>
+#include <Common/threadPoolCallbackRunner.h>
+
+namespace parquet
+{
+class RowGroupPageIndexReader;
+}
+
+namespace DB
+{
+class ParquetReader;
+struct FilterDescription;
+
+struct SelectResult
+{
+    std::optional<RowSet> set;
+    std::unordered_map<String, ColumnPtr> intermediate_columns;
+    IColumn::Filter intermediate_filter;
+    size_t valid_count = 0;
+    bool skip_all = false;
+};
+
+class SelectConditions
+{
+public:
+    SelectConditions(
+        std::unordered_map<String, SelectiveColumnReaderPtr> & readers_,
+        std::vector<String> & fast_filter_columns_,
+        std::vector<std::shared_ptr<ExpressionFilter>>& expression_filters,
+        const Block & header_);
+    SelectResult selectRows(size_t rows);
+private:
+    bool has_filter = false;
+    const std::unordered_map<String, SelectiveColumnReaderPtr> & readers;
+    const std::vector<String> & fast_filter_columns;
+    const std::vector<std::shared_ptr<ExpressionFilter>>& expression_filters;
+    const Block & header;
+};
+
+
+struct ColumnChunkData
+{
+    uint8_t * data;
+    size_t size;
+};
+
+struct ReadRangeBuffer
+{
+    arrow::io::ReadRange range;
+    PaddedPODArray<UInt8> buffer;
+};
+
+class RowGroupPrefetch;
+using RowGroupPrefetchPtr = std::shared_ptr<RowGroupPrefetch>;
```

Comment:
shared_ptr gives a false impression that the RowGroupPrefetch can be used after ParquetReader is destroyed (e.g. can be passed to a thread pool without strict cancellation). Use raw pointers instead.